src/WordPress/WordPressClientAPI.php

```diff
@@ -93,91 +93,99 @@ public function createPageRule($zoneId, $body)
     public function callAPI(Request $request)
     {
         $request = $this->beforeSend($request);
+        $response = $this->sendRequest($request);
+        $response = $this->getPaginatedResults($request, $response);
 
-        $url = add_query_arg($request->getParameters(), $this->getEndpoint().$request->getUrl());
+        return $response;
+    }
 
+
+    /**
+     * @param  Request $request
+     * @return [Array] $response
+     */
+    public function sendRequest(Request $request)
+    {
         $requestParams = array(
             'timeout' => 30,
             'method' => $request->getMethod(),
             'headers' => $request->getHeaders(),
         );
 
-        $isPaginatable = false;
-        if ($requestParams['method'] === 'GET') {
-            $isPaginatable = true;
+        if ($requestParams['method'] !== 'GET') {
+            $requestParams['body'] = json_encode($request->getBody());
+            $requestParams['headers']['Content-Type'] = 'application/json';
         }
 
-        $mergedResponse = null;
-
-        $currentPage = 1;
-        $totalPages = 1;
-
-        while ($totalPages >= $currentPage) {
-            // Enable pagination
-            if ($isPaginatable) {
-                $params['page'] = $currentPage;
```

Comment:
The bug was caused here and at line 97: `$url = add_query_arg($request->getParameters(), $this->getEndpoint().$request->getUrl());` function.

While refactoring to guzzle to missed `$params` variable which doesn't exist.

Other thing we missed was in Guzzle the args were suppose to be sent in `send` function where as in wordpress it's suppose to be sent in `add_query_arg`. We weren't setting the `page` parameter of the request so it wasn't grabbing pages after 1